src/inventory_client/inventory_client_test.go

```diff
@@ -0,0 +1,84 @@
+package inventory_client
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+func TestValidator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "inventory_test")
+}
+
+const (
+	testRetryDelay    = time.Duration(1) * time.Second
+	testRetryMaxDelay = time.Duration(2) * time.Second
+	testMaxRetries    = 3
+)
+
+var _ = Describe("inventory_client_tests", func() {
+	var (
+		clusterID = "cluster-id"
+		logger    = logrus.New()
+		client    *inventoryClient
+		server    *ghttp.Server
+	)
+
+	AfterEach(func() {
+		server.Close()
+	})
+
+	BeforeEach(func() {
+		var err error
+		server = ghttp.NewServer()
+		server.SetAllowUnhandledRequests(true)
+		server.SetUnhandledRequestStatusCode(http.StatusInternalServerError) // 500
+
+		client, err = CreateInventoryClientWithDelay(clusterID, "http://"+server.Addr(), "pullSecret", true, "",
+			logger, nil, testRetryDelay, testRetryMaxDelay, testMaxRetries)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(client).ShouldNot(BeNil())
+	})
+
+	Context("UpdateHostInstallProgress", func() {
+		var (
+			hostID = "host-id"
+		)
+
+		It("positive_response", func() {
+			expectServerCall(server, fmt.Sprintf("/api/assisted-install/v1/clusters/%s/hosts/%s/progress", clusterID, hostID), 200)
+			Expect(client.UpdateHostInstallProgress(hostID, models.HostStageInstalling, "")).ShouldNot(HaveOccurred())
+		})
+
+		It("negative_response", func() {
+			Expect(client.UpdateHostInstallProgress(hostID, models.HostStageInstalling, "")).Should(HaveOccurred())
+		})
+
+		It("positive_late_response", func() {
+			go func() {
+				time.Sleep(testRetryMaxDelay)
+				expectServerCall(server, fmt.Sprintf("/api/assisted-install/v1/clusters/%s/hosts/%s/progress", clusterID, hostID), 200)
```

Comment:
We should expect 2 calls, no?

Reply:
Unhandled requests receives 500 as described in BeforeEach section